docs/rfc/rfc-017-abci++-vote-extension-propag.md

```diff
@@ -0,0 +1,438 @@
+
+- **(d)** *Enough validators crash to block the rest*.
+
+    In this case, blockchain progress halts, i.e. surviving full nodes keep increasing rounds
+    indefinitely, until some of the crashed validators are able to recover.
+    Those validators that recover first will handshake with the Application and recover at the height
+    they crashed, which is still the same the blocked nodes are stuck in, so they don't need to catch up.
+    Further, they had persisted the extended commit for the previous height. Nothing to solve.
+
+    For those validators recovering later, we are in case (g) below.
+
+- **(e)** *Some validators crash, but not enough to block progress*.
+
+    When the correct processes that crashed recover, they handshake with the Application and resume at
+    the height they were at when they crashed. As the blockchain did not stop making progress, the
+    recovered processes are likely to have fallen behind with respect to the progressing majority.
+
+    At this point, the recovered processes are in case (g) below.
+
+- **(f)** *A new full node starts*.
+
+    The reasoning here also applies to the case when more than one full node are starting.
+    When the full node starts from scratch, it has no state (its current height is 0). Ignoring
+    statesync for the time being, the node just needs to catch up by applying past blocks one by one
+    (after verifying them).
+
+    Thus, the node is in case (g) below.
+
+- **(g)** *Advancing majority, lagging minority*
+
+    In this case, a set of full nodes, of which *n/3* or less are validators, fall behind an arbitrary
+    number of heights (e.g., temporary disconnection or network partition, memory thrashing, crashes,
+    new nodes).
+    These lagging nodes then need to catch up. They have to obtain the information they need to make
+    progress from other nodes. For each height *h*, this includes the decided block for *h*, and the
+    precommit votes also for *deciding h* (which can be extracted from the block at height *h+1*).
+    At a given height, say *h_c* a node will consider itself *caught up*, based on the (maybe out of
+    date) information it is getting from other nodes. Then, the node needs to be ready to propose
+    at height *h_c+1* which entails having received the vote extensions for *h_c*.
+
+    As the vote extensions are *not* stored in the blockchain, and it is difficult to have strong
+    guarantees on *when* a late node considers itself caught up, providing the late note with the right
+    vote extensions for the right height poses a problem.
+
+At this point, we have described and compared all cases raised in discussions leading up to this
+RFC. The list above aims at being exhaustive. The analysis of each case included above makes all of
+them converge into case (g).
+
+### Current Catch-up Mechanisms
+
+We now briefly describe the current catch-up mechanisms in the reactors concerned in Tendermint.
+
+#### Blocksync
+
+TODO
```

Comment:
We have a block sync spec in the works so perhaps we could link to that from here. Or keep a short summary with a link (so we do not loose references to the block sync in case if changes).

Reply:
Do you have a brief description (few sentences) we could set here, together with a link?
(and it the link is not ready, we can say "upcoming" or something)